src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGeneratorTest.java

```diff
@@ -302,10 +306,42 @@ public void testLastImport() {
         final int lineNumber = 8;
         final int columnNumber = 0;
         final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(rootAst, lineNumber,
-                columnNumber);
+                columnNumber, fileName);
         final List<String> actual = queryGenerator.generate();
         final List<String> expected = Collections.singletonList(
                 "/IMPORT[./DOT[@text='Iterator']]");
         assertEquals("Generated queries do not match expected ones", expected, actual);
     }
+
+    @Test
+    public void testTabWidth() throws Exception {
```

Comment:
add more exaples with tabs between, after tokens, with tabs which ends the line, etc. (Separate tests please)

Reply:
added three more tests